File: app/src/main/java/org/test/utils/LoadingAdapter.java
package org.test.utils;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.test.kotlin.demo.R;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class LoadingAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    @Override
    public int getItemCount() {
        return 1;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_loading, parent, false);

        return new RecyclerView.ViewHolder(view) {
        };
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
    }

}

File: app/src/main/java/org/test/utils/LoadingAdapter.kt
package org.test.utils

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import org.test.kotlin.demo.R

class LoadingAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun getItemCount() = 1

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = LayoutInflater
            .from(parent.context)
            .inflate(R.layout.item_loading, parent, false)
            .let(::LoadingHolder)

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {}

    class LoadingHolder(view: View) : RecyclerView.ViewHolder(view)

}
